fairroot/base/steer/FairRunSim.cxx

```diff
@@ -381,4 +382,32 @@ FairMCEventHeader* FairRunSim::GetMCEventHeader()
     return fMCEvHead;
 }
 
+void FairRunSim::ls(Option_t* option) const
+{
+    FairRun::ls(option);
+    TROOT::IncreaseDirLevel();
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+    if (fGen) {
+        fGen->ls(option);
+    }
+#pragma GCC diagnostic pop
+    if (fField) {
+        fField->ls(option);
+    }
+    // if (fSimulationConfig) {
+    //     fSimulationConfig->ls(option);
+    // }
+    ListOfModules->ls();
```

Comment:
I guess for the `TObjArray`s we should print something else?

```
OBJ: TObjArray	TObjArray	An array of objects : 0
```

is not that meaningful.

Reply:
Well, the `ls` member function of TObjArray is implemented by ROOT. So either we write our own free standing handler or improve the one in ROOT.

That said, we can start by giving our TObjArrays some name/title. So that at least the name/title is shown instead of `TObjArray … An array of objects`.

Reply:
How can you give it a name?

Reply:
[`TCollection::SetName`](https://root.cern/doc/v628/classTCollection.html#a6d6c3b2d795b309c0d8d2a7eebbb30d4)

I had hoped for `::SetTitle`. But that does not exist for TObjArray. So we can't set the title.

Reply:
Added a few `SetName()`.

See end of: https://cdash.gsi.de/testDetails.php?test=15932527&build=406173